docs/developers/guides/immutable.md

```diff
@@ -0,0 +1,474 @@
+
+```sh
+koinos-sdk-as-cli generate-contract-as --generate_authorize immutable.proto
+```
+``` { .txt, .no-copy }
+Generating boilerplate.ts and index.ts files...
+GENERATE_AUTHORIZE_ENTRY_POINT=1 yarn protoc --plugin=protoc-gen-as=./node_modules/.bin/koinos-as-gen --as_out=assembly/ assembly/proto/immutable.proto
+yarn run v1.22.22
+$ /home/$USER/Workspace/immutable/node_modules/.bin/protoc --plugin=protoc-gen-as=./node_modules/.bin/koinos-as-gen --as_out=assembly/ assembly/proto/immutable.proto
+@protobuf-ts/protoc installed protoc v26.1.
+Done in 5.07s.
+```
+
+In the following code snippets the highlighted lines were added by the previous invocation.
+
+```ts linenums="1" title="assembly/index.ts" hl_lines="11-20"
+import { System, Protobuf, authority } from "@koinos/sdk-as";
+import { Immutable as ContractClass } from "./Immutable";
+import { immutable as ProtoNamespace } from "./proto/immutable";
+
+export function main(): i32 {
+  const contractArgs = System.getArguments();
+  let retbuf = new Uint8Array(1024);
+
+  const c = new ContractClass();
+
+  switch (contractArgs.entry_point) {
+    case 0x4a2dbd90: {
+      const args = Protobuf.decode<authority.authorize_arguments>(
+        contractArgs.args,
+        authority.authorize_arguments.decode
+      );
+      const res = c.authorize(args);
+      retbuf = Protobuf.encode(res, authority.authorize_result.encode);
+      break;
+    }
+
+    case 0x2cf24dba: {
+      const args = Protobuf.decode<ProtoNamespace.hello_arguments>(
+        contractArgs.args,
+        ProtoNamespace.hello_arguments.decode
+      );
+      const res = c.hello(args);
+      retbuf = Protobuf.encode(res, ProtoNamespace.hello_result.encode);
+      break;
+    }
+
+    default:
+      System.exit(1);
+      break;
+  }
+
+  System.exit(0, retbuf);
+  return 0;
+}
+
+main();
+```
+
+```ts linenums="1" title="assembly/Immutable.boilerplate.ts" hl_lines="5-15"
+import { System, Protobuf, authority } from "@koinos/sdk-as";
+import { immutable } from "./proto/immutable";
+
+export class Immutable {
+  authorize(args: authority.authorize_arguments): authority.authorize_result {
+    // const call = args.call;
+    // const type = args.type;
+
+    // YOUR CODE HERE
+
+    const res = new authority.authorize_result();
+    res.value = true;
+
+    return res;
+  }
+
+  hello(args: immutable.hello_arguments): immutable.hello_result {
+    // const name = args.name;
+
+    // YOUR CODE HERE
+
+    const res = new immutable.hello_result();
+    // res.value = ;
+
+    return res;
+  }
+}
+```
+
+When we originally created this smart contract, we were given an example implementation of Hello World. Let's copy that code and move it to our new boiler plate file.
```

Comment:
Why are we copying in to the boiler plate file? Why not just copy the generated authorize function in to the original `immutable.ts`?

Reply:
Ah, because of imports. I think this might confuse people, but if they follow the guide as written, it will work.

Is it good to leave it this way, or tell people to copy the authorize function and imports?